### PR TITLE
tracee: fix pid file initialization

### PR DESCRIPTION
**Pull request by josedonizetti**
Fix https://github.com/aquasecurity/tracee/issues/2692

The pid initialization should be the last step before tracee blocks. I would actually prefer tracee lib itself didn't write this file, but instead, the binary did it, but for this we would have to refactor `Run` not to clear resources, moving the clearing to `Close`. 
So tracee would be executed like:

```
t.Run()
defer t.Close()
... (write pid, other things)
```

### CODE REVIEW

File: pkg/ebpf/tracee.go [hunk 1]
```diff
@@ -1370,6 +1356,13 @@ func (t *Tracee) Run(ctx gocontext.Context) error {
 	t.bpfLogsPerfMap.Start()
 	go t.processBPFLogs()
 	t.running = true
+
+	// write pid file
+	err = t.writePid()
+	if err != nil {
+		logger.Warn("memory dump", "error", err)
```
Comment (NDStrahilevitz) [inline pkg/ebpf/tracee.go:1363]: Error message seems wrong? Also I think we should log error and return here (with `t.Close()` called), not continue without the pid file.

Comment (geyslan) [reply pkg/ebpf/tracee.go:1363]: Yep, we need to abort.

**Review by geyslan — DISMISSED** [review]

File: pkg/ebpf/tracee.go (hunk 1 above)
Comment (josedonizetti) [reply pkg/ebpf/tracee.go:1363]: If CNDR depends on it, probably best to abort in case it was not able to write?

File: pkg/ebpf/tracee.go [hunk 2]
```diff
@@ -1370,8 +1356,17 @@ func (t *Tracee) Run(ctx gocontext.Context) error {
...
 	t.running = true
-	// block until ctx is cancelled elsewhere
-	<-ctx.Done()
+
+	// write pid file
+	err = t.writePid()
+	if err != nil {
+		// not able to write pid, abort
+		return err
+	} else {
```
Comment (NDStrahilevitz) [inline pkg/ebpf/tracee.go:1365]: nit: no need to `else` this as return will hit from `if`

**Review by rafaeldtinoco — APPROVED** [review]
LGTM

**Review by NDStrahilevitz — APPROVED** [review]
LGTM

**Comment by itaysk** [comment]
> I would actually prefer tracee lib itself didn't write this file, but instead, the binary did it, but for this we would have to refactor Run not to clear resources, moving the clearing to Close.
So tracee would be executed like:
t.Run()
defer t.Close()
... (write pid, other things)

I'm surprised this isn't the case today. should we open an issue for that?

**Comment by josedonizetti** [comment]
@itaysk yes, good point, the current strategy is required by CNDR, but we can find I different way of doing it that works for both places. I'll create the issue.

**Pull request merged**